cli/cmd/login.go

```diff
@@ -177,60 +177,78 @@ func runLogin(ctx context.Context, cmd *cobra.Command) (err error) {
 	if err != nil {
 		return fmt.Errorf("failed to create API client: %w", err)
 	}
 
-	currentTeam, err := config.GetValue("team")
-	if err != nil && !errors.Is(err, os.ErrNotExist) {
-		return fmt.Errorf("failed to get current team: %w", err)
+	// list all available teams
+	teams, err := cl.ListAllTeams(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to list teams: %w", err)
 	}
+
 	if cmd.Flags().Changed("team") {
-		selectedTeam := cmd.Flag("team").Value.String()
-		err = cl.ValidateTeam(ctx, selectedTeam)
+		// don't care about the current value
+		currentTeam := cmd.Flag("team").Value.String()
+		err = cl.ValidateTeamAgainstTeams(currentTeam, teams)
 		if err != nil {
-			return fmt.Errorf("failed to set team: %w", err)
+			return fmt.Errorf("failed to validate team: %w", err)
 		}
-		err = config.SetValue("team", selectedTeam)
+		err = config.SetValue("team", currentTeam)
 		if err != nil {
 			return fmt.Errorf("failed to set team: %w", err)
 		}
-	} else {
-		if currentTeam == "" {
-			// if current team is not set, try to set it from the API
-			teams, err := cl.ListAllTeams(ctx)
-			if err != nil {
-				return fmt.Errorf("failed to list teams: %w", err)
-			}
-			if len(teams) == 1 {
-				err = config.SetValue("team", teams[0])
-				if err != nil {
-					return fmt.Errorf("failed to set team: %w", err)
-				}
-				cmd.Printf("Your current team is set to %s.\n", teams[0])
-			} else {
-				cmd.Printf("Your current team is not set.\n\n")
-				cmd.Printf("Teams available to you: " + strings.Join(teams, ", ") + "\n\n")
-				cmd.Printf("To set your current team, run `cloudquery switch <team>`\n\n")
-			}
-		} else {
-			cmd.Printf("Your current team is set to %s.\n", currentTeam)
-		}
+		return nil
 	}
 
-	// Create a context for the shutdown with a 15-second timeout.
-	ctx, cancel := context.WithTimeout(ctx, 15*time.Second)
-	defer cancel()
-	if err := server.Shutdown(ctx); err != nil {
-		return fmt.Errorf("failed to shutdown server: %w", err)
+	currentTeam, err := config.GetValue("team")
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
+		return fmt.Errorf("failed to get current team: %w", err)
 	}
 
-	if serverErr != nil {
-		return serverErr
+	if len(currentTeam) > 0 && cl.ValidateTeamAgainstTeams(currentTeam, teams) == nil {
+		// The selected team is available to the user, so we're just using it
+		cmd.Printf("Your current team is set to %s.\n", currentTeam)
+		return nil
 	}
 
-	cmd.Println("CLI successfully authenticated.")
+	// either the team is not set, or the currently selected team is unavailable
+	if len(teams) == 1 {
+		currentTeam = teams[0]
+		err = config.SetValue("team", currentTeam)
+		if err != nil {
+			return fmt.Errorf("failed to set team: %w", err)
+		}
+		cmd.Printf("Your current team is set to %s.\n", currentTeam)
+		return nil
+	}
 
-	return nil
+	// 0 or > 1 teams available
+	cmd.Printf("Your current team is not set.\n\n")
+	cmd.Printf("Teams available to you: " + strings.Join(teams, ", ") + "\n\n")
```

Comment:
If the user doesn't have any teams we'll print an empty array. Maybe we should direct the user to create a team on the cloud Website instead?

Reply:
I think that would be a separate change unrelated to the fixed bug. I can do that as a follow-up